ops/internal/deployer/scripts/build-binaries.sh

```diff
@@ -0,0 +1,220 @@
+
+  # If download fails, fall back to building from source
+  echo "Pre-built binary not available. Falling back to building from source..."
+  if build_from_source "$full_version" "$binary_path"; then
+    echo "Successfully built and installed $stripped_version from source"
+    return 0
+  fi
+
+  echo "Failed to install $full_version"
+  return 1
+}
+
+try_download_release() {
+  local full_version=$1
+  local stripped_version=$2
+  local binary_path=$3
+
+  echo "Attempting to download $full_version for $OS-$ARCH from GitHub releases..."
+
+  # Construct GitHub release URL
+  local github_release_url="https://github.com/ethereum-optimism/optimism/releases/tag/$full_version"
+  echo "Checking release at $github_release_url"
+
+  # Extract version number without prefix for asset filename
+  local version_number=$(echo "$stripped_version" | sed 's/^v//')
+
+  # Construct asset filename based on OS and architecture with version number
+  local asset_name="op-deployer-${version_number}-${OS}-${ARCH}.tar.gz"
+  if [ "$OS" = "windows" ]; then
+    asset_name="op-deployer-${version_number}-${OS}-${ARCH}.zip"
+  fi
+
+  # Use the correct download URL format (keep the slash, don't URL encode it)
+  local download_url="https://github.com/ethereum-optimism/optimism/releases/download/$full_version/$asset_name"
+
+  # Download the archive file (silent failure)
+  echo "Downloading from $download_url"
+  if ! curl --fail -s -L -o "$TEMP_DIR/$asset_name" "$download_url"; then
+    echo "Release binary not found at $download_url"
+    return 1
+  fi
+
+  # Create extraction directory
+  local extract_dir="$TEMP_DIR/extract"
+  mkdir -p "$extract_dir"
+
+  # Extract the archive file
+  echo "Extracting archive..."
+  if [ "$OS" = "windows" ]; then
+    # Handle zip files for Windows
+    if ! unzip -q "$TEMP_DIR/$asset_name" -d "$extract_dir"; then
+      echo "Error: Failed to extract $TEMP_DIR/$asset_name"
+      return 1
+    fi
+  else
+    # Handle tar.gz for Linux and macOS
+    if ! tar -xzf "$TEMP_DIR/$asset_name" -C "$extract_dir"; then
+      echo "Error: Failed to extract $TEMP_DIR/$asset_name"
+      return 1
+    fi
+  fi
+
+  # Find the op-deployer binary in the extracted content
+  local binary_name="op-deployer"
+  if [ "$OS" = "windows" ]; then
+    binary_name="op-deployer.exe"
+  fi
+
+  # Check if binary exists in extract directory
+  if [ -f "$extract_dir/$binary_name" ]; then
+    local extracted_binary="$extract_dir/$binary_name"
+  else
+    # Try to find it recursively
+    local extracted_binary=$(find "$extract_dir" -name "$binary_name" -type f | head -n 1)
+    if [ -z "$extracted_binary" ]; then
+      echo "Error: Could not find $binary_name in extracted archive"
+      return 1
+    fi
+  fi
+
+  # Move to destination and make executable
+  echo "Installing binary to $binary_path"
+  mkdir -p "$(dirname "$binary_path")"
+  mv "$extracted_binary" "$binary_path"
+  chmod +x "$binary_path"
+
+  return 0
+}
+
+build_from_source() {
```

Comment:
Needed this complexity because we won't always have a release to download from. For example, devnets will run from tags that aren't finalized releases. Therefore we need to build from source, which then adds some complexity using mise to install the correct go version before building the binary.